### PR TITLE
[M1076] NumberInput: Don't try to cast null and undefined values

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[MR year and age values auto-filling as 0 (when should be left blank)](https://trello.com/c/5Qp7nNkZ/1076-mr-year-and-age-values-auto-filling-as-0-when-should-be-left-blank)

File: src/app/_mermaid_common/forms/directives/numberinput.dir.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ angular.module('mermaid.forms').directive('numberinput', [
 
         // Attempt to cast model val as number for view and validators
         modelCtrl.$formatters.push(function(modelValue) {
-          if (scope.ngModel != null) {
+          if (scope.ngModel != null && modelValue[attrs.widgetName] != null) {
```
Comment (saanobhaai) [inline src/app/_mermaid_common/forms/directives/numberinput.dir.js]: Will this cover `typeof modelValue[attrs.widgetName] === 'undefined'`?

Comment (gridcell) [reply src/app/_mermaid_common/forms/directives/numberinput.dir.js]: @saanobhaai Are you thinking `"undefined"` or `undefined`?

Comment (saanobhaai) [reply src/app/_mermaid_common/forms/directives/numberinput.dir.js]: er right `undefined` sans quotes

Comment (gridcell) [reply src/app/_mermaid_common/forms/directives/numberinput.dir.js]: the `!=` should capture undefined and null

**Pull request merged**